gramps/gen/filters/optimizer.py

```diff
@@ -78,11 +78,11 @@ def walk_filters(self, filter, parent_invert, result):
                         current_invert,
                         result,
                     )
-            elif hasattr(item, "map"):
-                rules.append(set(item.map))
+            elif hasattr(item, "selected_handles"):
+                rules.append(set(item.selected_handles))
         if rules:
             LOG.debug(
-                "filter %s: parent_invert=%s, invert=%s, op=%s, number of maps=%s",
+                "filter %s: parent_invert=%s, invert=%s, op=%s, number of selected_handles=%s",
```

Comment:
`selected_handles=%s`
You are printing `len(rules)` which does not, at face value, seem right